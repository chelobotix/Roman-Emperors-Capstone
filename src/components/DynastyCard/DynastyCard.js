import { Link } from "react-router-dom";
import style from "./DynastyCard.module.css";
import enter from "../../assets/images/enter.png";

const DynastyCard = (props) => {
  const { name, count, img } = props;

  return (
    <>
      <Link to={`/dynasty/${name}`} className={style.link}>
        <img src={enter} alt="enter" className={style.enter} />
      </Link>
      <img src={img} alt="Dynasty" className={style.img} />
      <h3 className={style.h3}>{name}</h3>
      <p className={style.p}>{count}</p>
    </>
  );
};

export default DynastyCard;
