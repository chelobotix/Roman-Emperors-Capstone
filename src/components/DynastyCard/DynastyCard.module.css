.link {
  align-self: flex-end;
}

.enter {
  width: 24px;
  margin: 5px;
}

.img {
  width: 100px;
  margin-top: 10px;
}

.h3 {
  color: #fff;
  align-self: flex-end;
  margin-right: 5px;
  margin-top: 10px;
}

.p {
  color: #fff;
  font-weight: bold;
  align-self: flex-end;
  margin-right: 5px;
}
