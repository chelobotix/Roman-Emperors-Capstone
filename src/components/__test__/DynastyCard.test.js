import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import DynastyCard from '../DynastyCard/DynastyCard';

const dynasty = {
  name: 'Gordian',
  count: 22,
};

describe('Home Rome Dynasty', () => {
  test('Get Dynasty Name', () => {
    render(
      <BrowserRouter>
        <DynastyCard
          name={dynasty.name}
          count={dynasty.count}
          img="emperor.jpg"
        />
      </BrowserRouter>,
    );
    const h3 = screen.getByRole('heading', { level: 3 });
    expect(h3).toHaveTextContent('Gordian');
  });

  test('Dynasty Card Tree', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <DynastyCard
          name={dynasty.name}
          count={dynasty.count}
          img="emperor.jpg"
        />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
