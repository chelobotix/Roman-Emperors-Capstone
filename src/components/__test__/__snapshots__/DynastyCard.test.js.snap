// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Rome Dynasty Dynasty Card Tree 1`] = `
Array [
  <a
    className="link"
    href="/dynasty/Gordian"
    onClick={[Function]}
  >
    <img
      alt="enter"
      className="enter"
      src="enter.png"
    />
  </a>,
  <img
    alt="Dynasty"
    className="img"
    src="emperor.jpg"
  />,
  <h3
    className="h3"
  >
    Gordian
  </h3>,
  <p
    className="p"
  >
    22
  </p>,
]
`;
